Sources/SwiftParser/Declarations.swift

```diff
@@ -544,26 +550,48 @@ extension Parser {
 }
 
 extension Parser {
+  @_spi(RawSyntax)
+  public mutating func parseMemberDeclListItem() -> RawMemberDeclListItemSyntax? {
+    let decl: RawDeclSyntax
+    if self.at(.poundSourceLocationKeyword) {
+      decl = RawDeclSyntax(self.parsePoundSourceLocationDirective())
+    } else {
+      decl = self.parseDeclaration()
+    }
+
+    let semi = self.consume(if: .semicolon)
+    var trailingSemas: [RawTokenSyntax] = []
```

Comment:
`Sema` -> `Semi`?

Could also just have this array and then use `trailingSemi.first` + `trailingSemi.dropFirst` below.

Reply:
Good catch with the typos. I prefer to consume the expected semicolon separately from the other ones though. Just a separation of normal parsing and recovery logic.